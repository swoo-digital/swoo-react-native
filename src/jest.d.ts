declare var renderWithStore: any
declare var wrapWithTheme: any
