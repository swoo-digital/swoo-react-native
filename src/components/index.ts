export { Button } from './button'
export { Typography } from './typography'
export { Counter } from './counter'
