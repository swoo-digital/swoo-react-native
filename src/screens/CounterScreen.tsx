import React from 'react';
import Box from '../components/Box';
import Counter from '../components/Counter';
import styled from 'styled-components/native';
import Button from '../components/Button';
import {useAppDispatch, useAppSelector} from '../redux/hooks';

const ScreenContainer = styled.SafeAreaView`
  flex: 1;
  justify-content: center;
  align-items: center;
  background: ${props => props.theme.colors.bg.navy};
`;

function CounterScreen() {
  const count = useAppSelector(state => state.counter.counter);
  const dispatch = useAppDispatch();

  const onPressAdd = () => {
    dispatch({type: 'counter/add'});
  };

  const onPressSubtract = () => {
    dispatch({type: 'counter/subtract'});
  };

  return (
    <ScreenContainer>
      <Box>
        <Button title="+1" onPress={onPressAdd} testID="addButton" />
        <Counter amount={count} testID="counter" />
        <Button title="-1" onPress={onPressSubtract} testID="subtractButton" />
      </Box>
    </ScreenContainer>
  );
}

export default CounterScreen;
