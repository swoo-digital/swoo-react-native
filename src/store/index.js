import {configureStore} from '@reduxjs/toolkit';
import counterReducer from '../store/features/counterSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
  },
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      immutableCheck: false,
      serializableCheck: false,
    }),
});
