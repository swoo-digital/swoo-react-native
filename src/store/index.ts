import { configureStore } from '@reduxjs/toolkit'
import type { TypedUseSelectorHook } from 'react-redux'
import { useDispatch, useSelector } from 'react-redux'

import persistedReducer from './rootReducer'

const store = configureStore({
  reducer: persistedReducer
})

const makeStore = () => store

type AppStore = ReturnType<typeof makeStore>
type AppDispatch = AppStore['dispatch']
type RootState = ReturnType<AppStore['getState']>

export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
