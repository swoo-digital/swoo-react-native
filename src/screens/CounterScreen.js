import React from 'react';
import {
  decreamentCounter,
  increamentCounter,
} from '../store/features/counterSlice';
import {useDispatch, useSelector} from 'react-redux';
import {Box} from '../components/Box';
import {Touchable} from '../components/Touchable';
import {BtnText, Text} from '../components/Text';
import {SafeAreaView} from 'react-native';
import {colors} from '../utils/colors';

const CounterScreen = () => {
  const dispatch = useDispatch();
  const {value} = useSelector(store => store.counter);

  return (
    <SafeAreaView style={{backgroundColor: colors.bg}}>
      <Box>
        <Touchable onPress={() => dispatch(increamentCounter())}>
          <BtnText>+1</BtnText>
        </Touchable>
        <Text>{value}</Text>
        <Touchable onPress={() => dispatch(decreamentCounter())}>
          <BtnText>-1</BtnText>
        </Touchable>
      </Box>
    </SafeAreaView>
  );
};

export default CounterScreen;
