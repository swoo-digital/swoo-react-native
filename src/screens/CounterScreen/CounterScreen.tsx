import React, {useState} from 'react';
import {Box} from '../../components/Box';
import {PressableButton} from '../../components/PressableButton';
import {Counter} from '../../components/Counter';

export const CounterScreen = () => {
  const [count, setCount] = useState(0);

  const increment = () => setCount(prev => prev + 1);

  const decrement = () => setCount(prev => prev - 1);

  return (
    <Box>
      <PressableButton title="- 1" onPress={increment} />
      <Counter>{count}</Counter>
      <PressableButton title="+ 1" onPress={decrement} />
    </Box>
  );
};
