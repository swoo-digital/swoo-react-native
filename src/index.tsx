import React from 'react';
import {View} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import styled, {useTheme} from 'styled-components';
import CounterButton from './components/CounterButton';
import {TextBold} from './components/Typography';
import Separator from './components/Separator';
import {RootState} from './store';
import {decrement, increment} from './store/reducers/counterReducer';

const Home = () => {
  const {colors} = useTheme();
  const dispatch = useDispatch();
  const counter = useSelector((state: RootState) => state.counter.value);

  const handleDecrease = () => {
    dispatch(decrement());
  };

  const handleIncrease = () => {
    dispatch(increment());
  };
  return (
    <Box>
      <CounterButton text="-1" onPress={handleDecrease} />
      <Separator size={104} />
      <TextBold fontSize={64} color={colors.green[0]}>
        {counter}
      </TextBold>
      <Separator size={104} />
      <CounterButton text="+1" onPress={handleIncrease} />
    </Box>
  );
};

export default Home;

const Box = styled(View)`
  background-color: ${({theme}) => theme.colors.blue[0]};
  flex: 1;
  align-items: center;
  justify-content: center;
`;
