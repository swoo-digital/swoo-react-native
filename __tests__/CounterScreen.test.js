import React from 'react';
import renderer from 'react-test-renderer';
import CounterScreen from '../src/screens/CounterScreen';

import {Provider} from 'react-redux';
import {store} from '../src/store';

it('counterScreen snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <CounterScreen />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
