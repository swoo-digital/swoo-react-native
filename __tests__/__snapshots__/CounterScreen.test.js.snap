// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counterScreen snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#004761",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#004761",
          "height": "100%",
          "justifyContent": "space-around",
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 24,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 160,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 24,
              "fontWeight": "bold",
            },
          ]
        }
      >
        +1
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#8CD6BD",
            "fontSize": 64,
            "fontWeight": "bold",
          },
        ]
      }
    >
      0
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 24,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 160,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 24,
              "fontWeight": "bold",
            },
          ]
        }
      >
        -1
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
